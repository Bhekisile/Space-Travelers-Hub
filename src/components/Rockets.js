import React from 'react';
import { useSelector } from 'react-redux';
import './styles/Rockets.css';

function Rockets() {
  const rockets = useSelector((state) => state.rocket);

  return (
    <div>
      <h2>Rockets</h2>
      <ul className="rocket-list">
        {rockets.map((rocket) => (
          <li key={rocket.id} className="rocket-item">
            <img src={rocket.flickr_images[0]} alt={rocket.name} className="rocket-image" />
            <div className="rocket-details">
              <h3>{rocket.name}</h3>
              <p>
                Type:
                {rocket.type}
              </p>
              {/* Add more details if needed */}
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default Rockets;
