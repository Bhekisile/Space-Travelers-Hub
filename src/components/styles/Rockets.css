.rocket-list {
  list-style: none;
  padding: 0;
}

.rocket-item {
  display: flex;
  margin-bottom: 20px;
  padding: 10px;
}

.rocket-image {
  width: 200px;
  height: 200px;
  margin-right: 20px;
}

.rocket-details {
  flex: 1;
}
