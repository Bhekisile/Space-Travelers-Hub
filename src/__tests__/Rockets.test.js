import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux'; // Import Provider
import configureStore from 'redux-mock-store';
import Rockets from '../components/Rockets';

const mockStore = configureStore([]);
const initialState = {
  rocket: [
    {
      id: 1,
      name: 'Falcon 9',
      description: 'Sample rocket description',
      flickr_images: ['image-url'],
      reserved: false,
    },
  ],
};
const store = mockStore(initialState);

test('Rockets component matches snapshot', () => {
  const { container } = render(
    <Provider store={store}>
      {' '}
      {/* Wrap with Provider */}
      <Router>
        <Rockets />
      </Router>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
