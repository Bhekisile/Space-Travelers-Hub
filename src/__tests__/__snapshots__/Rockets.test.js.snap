// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component matches snapshot 1`] = `
<div>
   
  <div>
    <ul
      class="rocket-list"
    >
      <li
        class="rocket-item"
      >
        <img
          alt="Falcon 9"
          class="rocket-image"
          src="image-url"
        />
        <div
          class="rocket-details"
        >
          <h3>
            Falcon 9
          </h3>
          <div>
            <p
              class="description"
            >
              Sample rocket description
            </p>
            <button
              class="Rbutton"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
